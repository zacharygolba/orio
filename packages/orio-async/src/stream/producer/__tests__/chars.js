// @flow

import Chars from '../chars'

const SOURCE = 'test'
let producer

beforeEach(() => {
  producer = new Chars(SOURCE)
})

test('#constructor()', () => {
  expect(producer).toMatchSnapshot()
})

test('#drop()', async () => {
  expect(producer.drop()).toBeUndefined()
  expect(await producer.next()).toMatchSnapshot()
  expect(producer).toMatchSnapshot()
})

test('#next()', async () => {
  expect(await producer.next()).toMatchSnapshot()
  expect(await producer.next()).toMatchSnapshot()
  expect(await producer.next()).toMatchSnapshot()
  expect(await producer.next()).toMatchSnapshot()
  expect(await producer.next()).toMatchSnapshot()
})
