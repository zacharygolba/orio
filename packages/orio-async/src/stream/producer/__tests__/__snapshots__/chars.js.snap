// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#constructor() 1`] = `
Chars {
  "cursor": 0,
  "source": "test",
}
`;

exports[`#drop() 1`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`#drop() 2`] = `
Chars {
  "cursor": 0,
  "source": "",
}
`;

exports[`#next() 1`] = `
Object {
  "done": false,
  "value": "t",
}
`;

exports[`#next() 2`] = `
Object {
  "done": false,
  "value": "e",
}
`;

exports[`#next() 3`] = `
Object {
  "done": false,
  "value": "s",
}
`;

exports[`#next() 4`] = `
Object {
  "done": false,
  "value": "t",
}
`;

exports[`#next() 5`] = `
Object {
  "done": true,
  "value": undefined,
}
`;
